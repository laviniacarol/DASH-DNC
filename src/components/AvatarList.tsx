import { StyledH2, StyledSpan } from '@/components'
import { Avatar, Box } from '@mui/material'
import { pxToRem } from '@/utils'
import { AvatarListProps } from '@/types'



function AvatarList(props: AvatarListProps) {
    return (
        <>
        {
          props.listData.map((item, index) => (
            <Box 
              key={index}
              sx={{
              alignItems: 'center',
              display: 'flex',
              padding: `${pxToRem(12)} 0`,
            }}>

         <Box>
         <Avatar
          alt={item.name}
          src={item.avatar}
          sx={{
            width: pxToRem(48),
            height: pxToRem(48),
            marginRight: pxToRem(16)
          }}
         />
         </Box>
          <Box>
            <StyledH2 color='inherit'>{item.name}</StyledH2>
            <StyledSpan color='inherit'>{item.subtitle}</StyledSpan>
          </Box>
        </Box>
          ))}
        </>
    )
}

export default AvatarList