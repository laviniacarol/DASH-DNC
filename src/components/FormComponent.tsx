import { FormComponentProps } from "@/types";

function FormComponent(props: FormComponentProps) {
    const { input, button, message } = props
    return (
        <form>
            {input.map((inputProps, index) => (
              <input key={index} {...inputProps} />
            ))}
             {button.map((buttonProps, index) => (
              <button key={index} {...buttonProps} />
            ))}
            {
                message && (<div style={{color: message.type === 'error' ? 'red' : 'green'}}>
                    {message.msg}
                </div>)
            }
        </form>
    )
}

export default FormComponent;