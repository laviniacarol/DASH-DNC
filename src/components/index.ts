export { default as CustomChart} from './CustomChart'
export { BannerImage } from './BannerImage'
export { default as CustomTable } from './CustomTable'
export * from './Logo'
export { default as AvatarList } from './AvatarList'
export { default as FormComponent} from './FormComponent'
export { default as Header} from './Header'
export * from './CardComponent'
export * from './StyleButton'
export * from './StyledInput'
export * from './Typographies'
