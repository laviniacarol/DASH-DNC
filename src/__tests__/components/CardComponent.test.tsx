import { CardComponent} from '@/components'
import { render } from '@testing-library/react'
import { DefaultTheme } from 'styled-components';
import { ThemeProvider } from 'styled-components'
import { themeList } from '../../resources/themesList'

describe('CardComponent', () => {
  const renderComponent = (theme: DefaultTheme,className?: string) =>
    render(
      <ThemeProvider theme={theme}>
        <CardComponent className={className}/>
      </ThemeProvider>
    )

  themeList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('it should match the spnapshot without any class', () => {
        const { asFragment } = renderComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })
        it('it should match the spnapshot with alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })
         it('it should match the spnapshot with success class', () => {
        const { asFragment } = renderComponent(theme, 'success')
        expect(asFragment()).toMatchSnapshot()
      })
        it('it should match the spnapshot with warning class', () => {
        const { asFragment } = renderComponent(theme, 'warning')
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})