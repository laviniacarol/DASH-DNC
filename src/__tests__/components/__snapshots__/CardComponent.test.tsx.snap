// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CardComponent Dark Theme it should match the spnapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #222;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`CardComponent Dark Theme it should match the spnapshot with success class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #222;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 success"
  />
</DocumentFragment>
`;

exports[`CardComponent Dark Theme it should match the spnapshot with warning class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #222;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`CardComponent Dark Theme it should match the spnapshot without any class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #222;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`CardComponent Light Theme it should match the spnapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`CardComponent Light Theme it should match the spnapshot with success class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 success"
  />
</DocumentFragment>
`;

exports[`CardComponent Light Theme it should match the spnapshot with warning class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`CardComponent Light Theme it should match the spnapshot without any class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #fdecea;
  border-color: #fdecea;
}

.c0.success {
  background-color: #e6ffed;
  border-color: #e6ffed;
}

.c0.warning {
  background-color: #fff4e5;
  border-color: #fff4e5;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
